sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2257,58 +2289,73 @@ def update_cluster_ips(
             exceptions.FetchIPError: if we failed to get the IPs. e.reason is
                 HEAD or WORKER.
         """
-
-        def is_provided_ips_valid(ips: Optional[List[Optional[str]]]) -> bool:
-            return (ips is not None and
-                    len(ips) == self.num_ips_per_node * self.launched_nodes and
-                    all(ip is not None for ip in ips))
-
-        use_internal_ips = self._use_internal_ips()
-
-        # cluster_feasible_ips is the list of IPs of the nodes in the cluster
-        # which can be used to connect to the cluster. It is a list of external
-        # IPs if the cluster is assigned public IPs, otherwise it is a list of
-        # internal IPs.
-        cluster_feasible_ips: List[str]
-        if is_provided_ips_valid(external_ips):
-            logger.debug(f'Using provided external IPs: {external_ips}')
-            cluster_feasible_ips = typing.cast(List[str], external_ips)
+        self.cached_cluster_info = cluster_info
+        if self.cached_cluster_info is None:
+            self._update_cluster_info()
```

Comment:
Is it possible that `self.cached_cluster_info is not None` before executing this function and we passed `cluster_info=None` into this function? IIUC in this case it will refresh the cluster info which incur higher execution time. Do we want to check if `self.cached_cluster_info is not None` before executing this?

Reply:
Oops, this is a good cache. We should check and use the `cluster_info` passed in with the argument and ignore the cached one, as we are using the external information to update the cached ips and infos.